src/openpgp.js

```diff
@@ -264,7 +264,10 @@ export function encrypt({ message, publicKeys, privateKeys, passwords, sessionKe
     if (privateKeys.length || signature) { // sign the message only if private keys or signature is specified
       message = await message.sign(privateKeys, signature, signingKeyIds, date, fromUserIds, message.fromStream, config);
     }
-    message = message.compress(config);
+    message = message.compress(
+      await getPreferredAlgo('compression', publicKeys || [], date, toUserIds, config),
```

Comment:
This is very nitpicky but maybe you could make `publicKeys` an optional argument, defaulting to `[]`, rather than including a default here?